File: src/main/java/com/github/unchama/seichiassist/commands/EventCommand.kt
package com.github.unchama.seichiassist.commands

import com.github.unchama.contextualexecutor.asNonBlockingTabExecutor
import com.github.unchama.messaging.EmptyMessage
import com.github.unchama.seichiassist.commands.contextual.builder.BuilderTemplates.playerCommandBuilder
import com.github.unchama.seichiassist.listener.new_year_event.NewYearBagListener
import com.github.unchama.seichiassist.listener.new_year_event.NewYearItemListener
import com.github.unchama.seichiassist.util.Util

object EventCommand {
  val executor = playerCommandBuilder
      .execution { context ->
        if (context.args.yetToBeParsed.firstOrNull() != "get") return@execution EmptyMessage

        val player = context.sender
        if (Util.isPlayerInventoryFull(player)) {
          Util.dropItem(player, NewYearBagListener.getNewYearBag())
          Util.dropItem(player, NewYearItemListener.getNewYearApple())
        } else {
          Util.addItem(player, NewYearBagListener.getNewYearBag())
          Util.addItem(player, NewYearItemListener.getNewYearApple())
        }

        return@execution EmptyMessage
      }
      .build()
      .asNonBlockingTabExecutor()
}

File: src/main/scala/com/github/unchama/seichiassist/commands/EventCommand.scala
package com.github.unchama.seichiassist.commands

import cats.effect.IO
import com.github.unchama.seichiassist.commands.contextual.builder.BuilderTemplates.playerCommandBuilder
import com.github.unchama.seichiassist.data.HalloweenItemData
import com.github.unchama.seichiassist.listener.new_year_event.{NewYearBagListener, NewYearItemListener}
import com.github.unchama.seichiassist.util.Util
import com.github.unchama.targetedeffect.TargetedEffect._
import org.bukkit.command.TabExecutor
import org.bukkit.entity.Player

object EventCommand {
  import com.github.unchama.targetedeffect._

  val newYearGrantEffect: TargetedEffect[Player] =
    Util.grantItemStacksEffect(
      NewYearBagListener.getNewYearBag,
      NewYearItemListener.getNewYearApple
    )

  val halloweenGrantEffect: TargetedEffect[Player] =
    Util.grantItemStacksEffect(
      HalloweenItemData.getHalloweenPotion
    )

  val executor: TabExecutor = playerCommandBuilder
    .execution { context =>
      val effect = context.args.yetToBeParsed match {
        case "newyear" :: _ => newYearGrantEffect
        case "halloween" :: _ => halloweenGrantEffect
        case _ => emptyEffect
      }

      IO.pure(effect)
    }
    .build()
    .asNonBlockingTabExecutor()
}